### PR TITLE
Add set property module

### PR DESCRIPTION
**Pull request by Josipmrden**
### Description

Add set property module to MAGE

### Pull request type

- [ ] Bugfix
- [X] Algorithm/Module
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

######################################

#### Module/Algorithm
- [X] Core algorithm/module implementation
- [X] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Unit tests
- [X] End-to-end tests
- [X] Code documentation
- [X] README short description
- [X] Documentation on [memgraph/docs](https://github.com/memgraph/docs)
- [ ] Update GQLALchemy signatures in [query builder](https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/graph_algorithms/query_builder.py  ) using [query module signature generator](https://github.com/memgraph/gqlalchemy/blob/main/scripts/query_module_signature_generator.py)

######################################

@kgolubic "Added set_property module inside Memgraph MAGE"
https://github.com/memgraph/documentation/pull/500

### CODE REVIEW

**Review by antepusic — APPROVED** [review]
All looks good (had to leave one suggestion :smile:)

**Review by antepusic — APPROVED** [review]
All should be good now :heavy_check_mark:

File: cpp/set_property_module/set_property_module.cpp [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+  }
+
+  const auto *maybe_username = std::getenv(kMgUsername);
+  if (maybe_username) {
+    username = std::move(maybe_username);
+  }
+
+  const auto *maybe_password = std::getenv(kMgPassword);
+  if (maybe_password) {
+    password = std::move(maybe_password);
```
Comment (antoniofilipovic) [inline cpp/set_property_module/set_property_module.cpp:51]: no need for std::move as you are copying pointer, and it is just variable copy

File: cpp/set_property_module/set_property_module.cpp [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+  }
+
+  return mg::Client::Params{.host = std::move(host),
+                            .port = std::move(port),
+                            .username = std::move(username),
+                            .password = std::move(password)};
+}
```
Comment (antoniofilipovic) [inline cpp/set_property_module/set_property_module.cpp:58]: mg::Client::Params accepts string and you are moving pointer.  I think correct solution is following
```suggestion
  return mg::Client::Params{.host = std::string(host),
                            .port = port,
                            .username = std::string(username),
                            .password = std::string(password)};
}
```

File: cpp/set_property_module/set_property_module.cpp [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+constexpr char *kMgPassword = "MG_PASSWORD";
+
+constexpr char *kDefaultHost = "localhost";
+constexpr uint16_t kDefaultPort = 7687;
+
+mg::Client::Params GetClientParams() {
+  auto *host = kDefaultHost;
+  auto port = kDefaultPort;
+  auto *username = "";
+  auto *password = "";
```
Comment (antoniofilipovic) [inline cpp/set_property_module/set_property_module.cpp:32]: You can just init struct here mg::Client::Params{...} and then update struct element if you find env variable

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Good work, I think GetClientParams is slightly off

**Comment by Josipmrden** [comment]
@antoniofilipovic  Do check when you have time!

**Review by antoniofilipovic — APPROVED** [review]
Looks good

**Pull request merged**